### ISSUE TITLE
Problem co-existing with New-Relic Infrastructure monitoring on RHEL

### ISSUE THREAD

**Issue by kmonsoor** [firehol/netdata#2144]
Hi there,  
In my organization, we run 10+ EC2 instances that are currently being monitored by `New-Relic Infrastructure` (NRI) SaaS product. I'm working on to replace it with a on-premise Netdata-InfluxDB-Grafana based solution, which is progressing fine.

However, we faced an interesting trouble. On an instance which has NRI agent is running, after installing Netdata and set up to send data to backend  properly, it works smoothly. But after couple of hours, the instance lose all connectivity with the outside "world".  
SSH fails to connect, AWS Health-monitor fails to monitor it; NRI stops sending data, Netdata stops sending data to backend, so complete dysfunctional. However, AWS console shows it running. So, it is turning into a jombie instance. Have to restart it.

After it happened on two instances, what we are doing is turning NRI off, when Netdata is being installed. Then it is running fine.

Any clue ?

**Notes**
*  The failed instances are both Redhat Enterprise Linux
*  Both are also running New-Relic APM product, baked in our Java-based product; which is having no issues with Netdata
*   `memory mode` was set to `save`
*  After restart, as I've checked the `error.log` of Netdata, it didn't show anything interesting.

**Comment by ktsaou** [comment]
hm... it is interesting....

Did you force an unclean shutdown? If not, netdata should have saved its charts. Did they show anything unusual?

I am not aware of NRI. Is it capable of provisioning things on the machine? The problem you face looks like something is changing the routing table or the firewall of machine. What if you run just NRI? Is the server stable then?
 
In general, netdata is a read only process. It is not capable of making changes to the operating system. Also, by default it runs with a redicously low priority (process scheduler set to IDLE, so that netdata gets CPU only when nothing else wants it - it is lower than nice 19), it sets its out-of-memory score to 1000 (so that netdata will be the first to be killed if the system runs out of memory), and runs completely unpriviliged (user `netdata` - exception is `apps.plugin` that has special capabilities to read disk I/O metrics for all processes from `/proc`).

My suggestion is first to verify that the server is alive during this period. If the server is not alive, then something is crashing it. Theoretically speaking, since netdata reads a lot of files in `/proc`, and possibly NRI is doing something similar too, there could be a kernel issue that appears only when 2 different programs are reading something in parallel from proc (I doubt this is the case though - but anyway it could be).

If you can experiment a bit, you can verify netdata is not interfering somehow with NRI, by limiting the netdata modules running. Fetch netdata.conf from `http://your.server.ip:19999/netdata.conf` and find the `[plugins]` section. You will get something like this:

```
[plugins]
	# tc = yes
	# idlejitter = yes
	# proc = yes
	# diskspace = yes
	# cgroups = yes
	# checks = no
	# enable running new plugins = yes
	# check for new plugins every = 60
	# charts.d = yes
	# apps = yes
	# node.d = yes
	# fping = yes
	# python.d = yes
```

set all plugins to `no` (uncomment the lines too), except `proc`. This will give you a netdata with basic system monitoring. Is it stable with just that? If it is, you can add more modules until it appears again.

There is another section for `proc` submodules, in case you want to limit even the `proc` module:

```
[plugin:proc]
	# netdata server resources = yes
	# /proc/stat = yes
	# /proc/uptime = yes
	# /proc/loadavg = yes
	# /proc/sys/kernel/random/entropy_avail = yes
	# /proc/interrupts = yes
	# /proc/softirqs = yes
	# /proc/vmstat = yes
	# /proc/meminfo = yes
	# /sys/kernel/mm/ksm = yes
	# /sys/devices/system/edac/mc = yes
	# /sys/devices/system/node = yes
	# /proc/net/dev = yes
	# /proc/net/netstat = yes
	# /proc/net/snmp = yes
	# /proc/net/snmp6 = yes
	# /proc/net/softnet_stat = yes
	# /proc/net/ip_vs/stats = yes
	# /proc/net/stat/conntrack = yes
	# /proc/net/stat/synproxy = yes
	# /proc/diskstats = yes
	# /proc/net/rpc/nfsd = yes
	# /proc/net/rpc/nfs = yes
	# /proc/spl/kstat/zfs/arcstats = yes
	# ipc = yes
```

**Comment by kmonsoor** [comment]
Hi @ktsaou thanks for your reply. I intentionally took some time to reply to add more contextual information.

looks like, I managed to "freeze" up another RHEL instance; but this time it was NOT running any NRI process. Netdata on it was able to send stats to Influx for around 30 seconds after final configuration and restarting the service. So, it seems trouble with RHEL rather than with NRI.

I had used the all-inclusive command to "prepare" the system before installation.
```
curl -Ss 'https://raw.githubusercontent.com/firehol/netdata-demo-site/master/install-required-packages.sh' >/tmp/kickstart.sh && bash /tmp/kickstart.sh -i netdata-all
```

I'm trying to digging in more and I'll keep this thread posted on any update of the situation.

**Comment by ktsaou** [comment]
hm... this is even more interesting... an application should never be able to freeze a kernel... and netdata does not do anything unusual to trigger or influence a kernel freeze...

So, yes please, if you can trace this, please do so...

**Comment by kmonsoor** [comment]
However, if it worth mentioning, my conf file was including following options 

```
history = 10,800
...
[plugins]
checks = yes
...
[backend]
	enabled = yes
	data source = average
	type = graphite
	destination = x.a.y.b:2003
       hostname = xa-yb
```

**Comment by ktsaou** [comment]
I don't see anything unusual. It should work perfectly...

**Comment by ktsaou** [comment]
although the `checks` plugin is pretty useless for you. It is a plugin I need to find time differences between various plugins and timing methods...

**Comment by kmonsoor** [comment]
looks like, our instances were too outdated and/or system reboot were pending. Because, each of them are running perfectly after the "freeze ➡️restart". 

Interesting to note that my POC installation were on a similarly insanely  outdated Ubuntu 14.04, on which Netdata+Influx+Grafana is running smoothly till date.  

i guess someone may ask why they aren't updated regularly. I can't answer that as I'm not the devOps here. Maybe there's some wisdom behind that.

**Comment by kmonsoor** [comment]
so, @ktsaou, if i may suggest, I recommend that RHEL installation section might include a note mentioning that outdated systems might need a reboot.

**Comment by ktsaou** [comment]
nice! 

I guess the installation page should be more appropriate. But there is no RHEL specific section in it. You may add a note though...
